pkg/controller/flink/client/api.go

```diff
@@ -142,16 +142,19 @@ func (c *FlinkJobManagerClient) GetClusterOverview(ctx context.Context, url stri
 }
 
 // Helper method to execute the requests
-func (c *FlinkJobManagerClient) executeRequest(
+func (c *FlinkJobManagerClient) executeRequest(ctx context.Context,
 	method string, url string, payload interface{}) (*resty.Response, error) {
+	client := resty.SetLogger(logger.GetLogWriter(ctx)).SetTimeout(defaultTimeOut)
+
 	var resp *resty.Response
 	var err error
 	if method == httpGet {
-		resp, err = c.client.R().Get(url)
+		client.SetTimeout(httpGetTimeOut).SetRetryCount(retryCount)
```

Comment:
Out of curiosity, why two separate timeouts?

Reply:
More discussion here: https://github.com/lyft/flinkk8soperator/pull/9

Main reason is this is a blocking call, and Get calls generally finishes in milliseconds. It's only the POST request in Flink which causes issues under low CPU, and POST is not idempotent.

Problem with using the same timeout for Get calls (like 1 minute timeout) is when the cluster is starting, and pods are not ready, all the requests are going timeout as there is no job manager to serve the requests. This will just keep all the workers blocked. It is better to give up, and poll in the next resync period.

Reply:
Cool, thanks for the context :) The difference in timeouts makes sense. Given that this would depend on some cluster specific resources, should we make this configurable (with some sensible defaults as you have here) ?

Reply:
@glaksh100 Its actually the other way. You will have all kinds of behavior based in application configuration/resources. Intention is to get to a value which would not affect the cycles of the operator. Also if there is a better number for timeout it’s better to find and set that value.